### DISCUSSION TITLE
Feature Request: Recursively Foreach

### DISCUSSION THREAD

**Discussion by wisamidris7** [dotnet/csharplang#8016] [General]
I know this is looks crazy idea but it will be very good for c# cause in dart now it contains recursively foreach
So can c# have recursively foreach to make it very easy

It could be like this

```c#
foreach (var item in Items; var level = 0; var children = item.Children)
{
     Console.WriteLine("We are in level " + level);
     Console.WriteLine("We are in item" + item.Name);
     Console.WriteLine("We are in my children count" + children?.Count() ?? 0);
}
```

This is the best thing i think with if you can make a different or easier one it's okay but that what i think will be awesome

**Comment by wisamidris7** [comment]
Okay how it will be?

**Comment by HaloFour** [comment]
It's pretty trivial to write an extension method to support this, which can then be composed with LINQ.

**Reply by wisamidris7** [reply]
Yes this is also good

**Comment by mattwar** [comment]
What happens when item has more than one collection member that you would like to recurse into?

**Reply by jnm2** [reply]
Presumably, `.Concat`?
